hypothesis-python/src/hypothesis/extra/numpy.py

```diff
@@ -846,3 +846,65 @@ def broadcastable_shapes(shape, min_dims=0, max_dims=None, min_side=1, max_side=
         min_side=min_side,
         max_side=max_side,
     )
+
+
+@st.defines_strategy
+def integer_array_indices(shape, result_shape=array_shapes(), dtype="int"):
+    # type: (Tuple[int, ...], SearchStrategy[Tuple[int, ...]], Union[np.dtype, str]) -> st.SearchStrategy[Tuple[np.ndarray, ...]]
```

Comment:
We tend to make the type hints err on the side of strictness, so `np.dtype` instead of `Union[np.dtype, str]` as most strings are not valid here.